gobblin-azkaban/src/main/java/gobblin/azkaban/AzkabanJobLauncher.java

```diff
@@ -125,12 +144,41 @@ public void run()
   public void cancel()
       throws Exception {
     try {
-      this.jobLauncher.cancelJob(this.jobListener);
+      cancelJob(this.jobListener);
     } finally {
-      this.closer.close();
+      try {
+        stop();
+      } finally {
+        close();
+      }
     }
   }
 
+  @Override
+  public void start() throws ApplicationException {
+    this.applicationLauncher.start();
+  }
+
+  @Override
+  public void stop() throws ApplicationException {
+    this.applicationLauncher.stop();
```

Comment:
The same as above.  If this was done, job developers could also have resources that are initialized once at job start and cleaned up at job end.